src/pkg/packager/actions.go

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains functions for interacting with, managing and deploying zarf packages.
+package packager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/defenseunicorns/zarf/src/internal/packager/template"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/exec"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Run commands that a component has provided.
+func (p *Packager) runAction(actionSet types.ZarfComponentActionSet, actions []types.ZarfComponentAction, valueTemplate *template.Values) error {
```

Comment:
IMO it looks a lot clear to replace the `actionSet types.ZarfComponentActionSet` parameter with `actionSetDefaults types.ZarfComponentActionDefaults`.

That's all we're using the actionSet for and that way we're passing a struct and one of its own member fields to a singular function 😛

Reply:
Yeah that seems reasonable, I thought I might use more than that initially, but looks like I didn't.